### ISSUE TITLE
QFieldSync crashes when when layer properties are open on Windows

### ISSUE THREAD

**Issue by suricactus** [opengisch/qfieldsync#310]
Hello,
I downloaded the latest QFIELDSYNC vesion 4.0.0-Beta-12
It now provokes a QGIS Crash when I try to open the layer properties.
I uninstalled QFIELDSYNC and now can open the layer properties without any crash.
Crash ID: c5db59a4c6e52b8a7290a70f4481b2819dc7774b

```
Stack Trace

QObject::parent :
QLayout::addChildWidget :
QStackedLayout::insertWidget :
QgsVectorLayerProperties::addPropertiesPageFactory :
QgisApp::showLayerProperties :
QgisApp::layerProperties :
QMetaObject::activate :
QAction::activate :
QMenu::actionGeometry :
QMenu::actionGeometry :
QMenu::mouseReleaseEvent :
QWidget::event :
QMenu::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QMenu::exec :
QgsLayerTreeView::contextMenuEvent :
QWidget::event :
QFrame::event :
QAbstractItemView::viewportEvent :
QCoreApplicationPrivate::sendThroughObjectEventFilters :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :




QGIS Info
QGIS Version: 3.16.4-Hannover
QGIS code revision: 654e76b3cb
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.1.4
Running against GDAL: 3.1.4



System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.18363
```
_Originally posted by @jpdupuy in https://github.com/opengisch/qfieldsync/issues/307#issuecomment-918014117_

**Comment by suricactus** [comment]
Hey, @jpdupuy thanks for reporting this problem. We have tried to reproduce the problem on two different windows machines and didn't manage. Is it constantly behaving like this or there are certain steps that you need to do to see this error?

**Comment by jpdupuy** [comment]
Hi, It is constantly behaving like this. 
I do not know if it is library dependant, because I also installed QGIS 3.20
I installed and unstalled the plugin several times, and could observe the same result.

**Comment by suricactus** [comment]
Thanks for the feedback, will continue to search for the cause of the error on your machine.

**Comment by jpdupuy** [comment]
Might be of some interest to you.
It does not crashes QGIS 3.20 when I open the layer properties

**Comment by suricactus** [comment]
Quite interesting one. The problem has been detected and solved and it is on this very specific 3.16.4 version. The python version 3.16.4 is packaged with older Python compared to 3.16.11 for example. There might be other issues that you encounter too. Fix arriving with the new beta.

**Comment by suricactus** [comment]
Should have been fixed in BETA15.
